app/src/main/java/com/sourcepoint/test_project/MainActivity.java

```diff
@@ -41,6 +41,8 @@ private CCPAConsentLib buildAndRunConsentLib(Boolean showPM) throws ConsentLibEx
     protected void onResume() {
         super.onResume();
         try {
+            if (ccpaConsentLib != null)
```

Comment:
Instead of destroying the instance of consent lib we can simply not have one at all! Just like we are doing on the GDPR example app one can call the run() method without saving the consent lib instance in a field. This way way solve this very issue and some memory leaks along side :)